Gems/ScriptCanvas/Code/Include/ScriptCanvas/Grammar/AbstractCodeModel.cpp

```diff
@@ -2957,6 +2970,10 @@ namespace ScriptCanvas
                     m_activeDefaultObject.insert(&node);
                 }
             }
+            else if (auto functionCallNode = azrtti_cast<const Nodes::Core::FunctionCallNode*>(&node))
```

Comment:
this branch is kind of weird, as first two branches are for nodeable and function, so when reaching this check, it won't be function?

Reply:
View in context. This is only about node dependencies. Its ok for all other nodes not to return an interface, but an error if FunctionCallNodes do not.

Reply:
as previous branch already catch this case 
```
auto functionCallNode = azrtti_cast<const Nodes::Core::FunctionCallNode*>(&node)
```
so I don't think this branch can be reached with this check again unless I read the check wrong? (because it is if else statement, not switch)

Reply:
```void AbstractCodeModel::ParseDependencies(const Node& node)
        {
            const auto dependencyOutcome = node.GetDependencies();

            if (dependencyOutcome.IsSuccess())
            {
                const auto& dependencies = dependencyOutcome.GetValue();

                // #functions2 this search needs to recurse, this layer of dependencies will only be one step deep
                // currently this problem is found by the asset processor
                if (dependencies.userSubgraphs.find(m_source.m_namespacePath) != dependencies.userSubgraphs.end())
                {
                    AZStd::string circularDependency = AZStd::string::format
                    (ParseErrors::CircularDependencyFormat
                        , m_source.m_name.data()
                        , node.GetDebugName().data()
                        , m_source.m_name.data());

                    AddError(nullptr, aznew Internal::ParseError(node.GetEntityId(), circularDependency));
                }

                // #functions2 make this use an identifier for the node, for property window display and easier find/replace updates
                // this part must NOT recurse, the dependency tree should remain a tree and not be flattened
                m_orderedDependencies.source.MergeWith(dependencies);
            }
            else
            {
                AddError(nullptr, ValidationConstPtr(aznew DependencyRetrievalFailiure(node.GetEntityId())));
            }

            if (auto subgraphInterface = node.GetSubgraphInterface())
            {
                m_subgraphInterface.MergeExecutionCharacteristics(*subgraphInterface);

                if (subgraphInterface->HasOnGraphStart())
                {
                    m_subgraphStartCalls.insert(&node);
                }

                if (subgraphInterface->IsActiveDefaultObject())
                {
                    m_activeDefaultObject.insert(&node);
                }
            }
            else if (auto functionCallNode = azrtti_cast<const Nodes::Core::FunctionCallNode*>(&node))
            {
                AddError(node.GetEntityId(), nullptr, "FunctionCallNode failed to return latest SubgraphInterface");
            }
        }```

Reply:
I'm confused about what case you are talking about

Reply:
nvm, it seems like I read the wrong check!